### PR TITLE
Desktop 3.4.0 Release notes

### PR DESCRIPTION
**Pull request by mat007**
### Proposed changes

Added Docker Desktop 3.4.0 release notes.

### CODE REVIEW

File: docker-for-mac/release-notes.md [hunk 1]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
-## Docker Desktop 3.3.3
-2021-05-06
+## Docker Desktop 3.4.0
+2021-06-07
 
 [Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
 [Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}
 
-## Upgrades
+### New
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:22]: Should we add a note on Volume Management and Compose canary roll out here?

File: docker-for-mac/release-notes.md [hunk 2]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+- [Go 1.16.5](https://github.com/golang/go/releases/tag/go1.16.5)
+
+### Bug fixes and minor changes
+
+- Prevent `docker run` from hanging if inotify event injection fails. Fixes [docker/for-mac#5590](https://github.com/docker/for-mac/issues/5590).
+- Fixed error showing stderr log in the UI. Fixes [docker/for-mac#5688](https://github.com/docker/for-mac/issues/5688).
+- Fix emulation of `riscv64`. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).
+- Automatically reclaim space after deleting containers by deleting volumes and removing build cache.
+- Respect the bypass setting in the manual proxy configuration when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
+- Fixed Docker Compose application removal from the UI which with custom file name other than `docker-compose.yml`. Fixes [docker/for-win#11046](https://github.com/docker/for-win/issues/11046)
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:39]: How about " Fixed an issue which caused Docker Compose application to be removed from Docker Desktop if it contains a file name other than `docker-compose.yml`."

Comment (mat007) [reply docker-for-mac/release-notes.md:39]: It’s actually the other way around I believe: the application couldn’t be removed.

File: docker-for-mac/release-notes.md (hunk 1 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:22]: Ah yes, indeed. I’ll get some content from the blog post draft.

File: docker-for-mac/release-notes.md [hunk 3]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+- Expose the host CPU on Apple Silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:40]:
```suggestion
- Docker Desktop now exposes the host CPU on Apple silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).
```

File: docker-for-mac/release-notes.md (hunk 2 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:39]: Aha but indeed that sentence doesn’t make any sense 😀 
How about «Fixed deleting Docker Compose applications not removing from the UI when based on a file name different from `docker-compose.yml`.»?

File: docker-for-mac/release-notes.md (hunk 1 above)
Comment (stephen-turner) [reply docker-for-mac/release-notes.md:22]: Yes. And what about the update notifications changes? (cc @dieucao)

File: docker-for-mac/release-notes.md [hunk 4]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+- [Kubernetes 1.21.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.1)
+- [containerd v1.4.6](https://github.com/containerd/containerd/releases/tag/v1.4.6)
+- [runc v1.0.0-rc95](https://github.com/opencontainers/runc/releases/tag/v1.0.0-rc95)
+- [Go 1.16.5](https://github.com/golang/go/releases/tag/go1.16.5)
+
+### Bug fixes and minor changes
+
+- Prevent `docker run` from hanging if inotify event injection fails. Fixes [docker/for-mac#5590](https://github.com/docker/for-mac/issues/5590).
+- Fixed error showing stderr log in the UI. Fixes [docker/for-mac#5688](https://github.com/docker/for-mac/issues/5688).
+- Fix emulation of `riscv64`. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:36]:
```suggestion
- Fixed an issue which caused `riscv64` emulation to fail on Docker Desktop. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).
```

File: docker-for-mac/release-notes.md [hunk 5]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+- Automatically reclaim space after deleting containers by deleting volumes and removing build cache.
+- Respect the bypass setting in the manual proxy configuration when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:38]:
```suggestion
- Docker Desktop now respects the bypass proxy setting through the **Manual proxy configuration** option when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
```

Comment (usha-mandya) [reply docker-for-mac/release-notes.md:38]: Is it fixed both in the UI and the CLI?

File: docker-for-mac/release-notes.md [hunk 6]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+- Fixed Docker Compose application removal from the UI which with custom file name other than `docker-compose.yml`. Fixes [docker/for-win#11046](https://github.com/docker/for-win/issues/11046)
+- Expose the host CPU on Apple Silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).
+- Avoid leaking open ports bound to privileged ports and specific IPs over engine Restart. Fixes [docker/for-mac#5649](https://github.com/docker/for-mac/issues/5649).
+- Use `vpnkit` with `virtualization.framework` to fix connectivity issues with VPN clients such as Cisco AnyConnect.
+- Fixed version number missing in update dialog window.
+- Fixed diagnostics sometimes not uploading correctly from the support dialog window.
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:44]:
```suggestion
- Fixed an issue where the diagnostics were sometimes not uploaded correctly from the **Support** dialog.
```

File: docker-for-mac/release-notes.md (hunk 1 above)
Comment (usha-mandya) [reply docker-for-mac/release-notes.md:22]: Yes, totally forgot about that. Good point. I believe we should mention the change to update notification, but @dieucao's call :)

File: docker-for-mac/release-notes.md [hunk 7]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+
+## Docker Desktop 3.3.3
+2021-05-06
+
+[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:49]:
```suggestion
[Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64)
```

Comment (usha-mandya) [reply docker-for-mac/release-notes.md:49]: Suggest removing the primary button here so we emphasize only the latest release

File: docker-for-mac/release-notes.md [hunk 8]
```diff
@@ -13,35 +13,65 @@ This page contains information about the new features, improvements, known issue
...
+[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:50]:
```suggestion
[Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/64133/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64)
```

File: docker-for-mac/release-notes.md (hunk 2 above)
Comment (usha-mandya) [reply docker-for-mac/release-notes.md:39]: Aah OK. Thanks for clarifying. I thought the application was automatically removed if it contained a custom file name :D 

Flipped the order to see if this reads better. "Docker Compose applications with file names other than `docker-compose.yml` can now be removed from Docker Desktop."

File: docker-for-mac/release-notes.md (hunk 8 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:50]: Are you sure about that? Should I remove the buttons for all the old releases as well?

File: docker-for-mac/release-notes.md (hunk 5 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:38]: I don’t know, @djs55 ?

File: docker-for-mac/release-notes.md (hunk 7 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:49]: Oh sorry, that answers my other question then. Thanks!

File: docker-for-mac/release-notes.md (hunk 2 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:39]: Yes, that’s better. Thanks!

File: docker-for-mac/release-notes.md (hunk 7 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:49]: Should we not add one for the Windows release notes as well, by the way?

File: docker-for-mac/release-notes.md (hunk 1 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:22]: I grabbed selected sentences from the blog post. I’m not sure the joyful tone is right for release notes, though. 😁

File: docker-for-mac/release-notes.md [hunk 9]
```diff
@@ -13,35 +13,73 @@ This page contains information about the new features, improvements, known issue
...
+## Docker Desktop 3.4.0
+2021-06-07
 
 [Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-amd64){: .button .primary-btn}
 [Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/Docker.dmg?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-mac-arm64){: .button .primary-btn}
 
-## Upgrades
+### New
+
+In this release we’re introducing a new capability in Docker Desktop that helps you to create and delete volumes from Desktop’s Dashboard as well as to see which ones are In Use.
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:24]:
```suggestion
**Volume Management**: Docker Desktop users can now create and delete volumes using the Docker Dashboard and also see which volumes are being used. For more information, see [Explore volumes](../desktop/dashboard.md#explore-volumes).

**Compose V2 beta**: Docker Desktop now includes the beta version of Compose V2, which supports the 'compose' command as part of the Docker CLI. For more information, see [Compose V2 beta](../compose/cli-command.md).

**Skip Docker Desktop updates**:  All users can now skip an update when they are prompted to install individual Docker Desktop releases. For more information, see [Docker Desktop updates](../docker-for-windows/install.md#updates).
```

Comment (mat007) [reply docker-for-mac/release-notes.md:24]: Thanks!

File: docker-for-mac/release-notes.md [hunk 10]
```diff
@@ -13,35 +13,77 @@ This page contains information about the new features, improvements, known issue
...
+
+**Volume Management**: Docker Desktop users can now create and delete volumes using the Docker Dashboard and also see which volumes are being used. For more information, see [Explore volumes](../desktop/dashboard.md#explore-volumes).
+
+**Compose V2 beta**: Docker Desktop now includes the beta version of Compose V2, which supports the 'compose' command as part of the Docker CLI. For more information, see [Compose V2 beta](../compose/cli-command.md).
+
+**Skip Docker Desktop updates**: All users can now skip an update when they are prompted to install individual Docker Desktop releases. For more information, see [Docker Desktop updates](../docker-for-windows/install.md#updates).
+
+### Deprecation
+
+- Docker Desktop no longer installs Notary, `docker trust` should be used for image signing.
```
Comment (mat007) [inline docker-for-mac/release-notes.md:32]: @usha-mandya ☝️

File: docker-for-windows/release-notes.md [hunk 11]
```diff
@@ -13,17 +13,55 @@ This page contains information about the new features, improvements, known issue
...
+
+### Bug fixes and minor changes
+
+- Fixed error showing stderr log in the UI. Fixes [docker/for-win#11251](https://github.com/docker/for-win/issues/11251).
+- Automatically reclaim space after deleting containers by deleting volumes and removing build cache.
+- Docker Compose applications with file names other than `docker-compose.yml` can now be removed from Docker Desktop. Fixes [docker/for-win#11046](https://github.com/docker/for-win/issues/11046)
+- Fixed version number missing in update dialog window.
+- Fixed an issue where the diagnostics were sometimes not uploaded correctly from the **Support** dialog.
+- Fixed DNS entries for `*.docker.internal` and Kubernetes cluster reset after the VM IP changes.
+- Fixed corrupted internal cache preventing Docker Desktop from starting. Fixes [docker/for-win#8748](https://github.com/docker/for-win/issues/8748).
```
Comment (mat007) [inline docker-for-windows/release-notes.md:52]: @usha-mandya I added that one as well, I just found out that the related PR did not have the proper marking to list it.

File: docker-for-mac/release-notes.md [hunk 12]
```diff
@@ -13,35 +13,79 @@ This page contains information about the new features, improvements, known issue
...
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:32]:
```suggestion
- Docker Desktop no longer installs Notary. You can now use [Docker Content Trust](../engine/security/trust.md) for image signing.
```

File: docker-for-mac/release-notes.md (hunk 7 above)
Comment (usha-mandya) [reply docker-for-mac/release-notes.md]: Sorry, missed this somehow.

> Should we not add one for the Windows release notes as well, by the way?

Good question. We can add it only to the latest release for consistency.

File: docker-for-windows/release-notes.md (hunk 11 above)
Comment (usha-mandya) [inline docker-for-windows/release-notes.md:52]:
```suggestion
- Fixed a corrupted internal cache which was preventing Docker Desktop from starting. Fixes [docker/for-win#8748](https://github.com/docker/for-win/issues/8748).
```

File: docker-for-mac/release-notes.md [hunk 13]
```diff
@@ -13,35 +13,79 @@ This page contains information about the new features, improvements, known issue
...
+- [runc v1.0.0-rc95](https://github.com/opencontainers/runc/releases/tag/v1.0.0-rc95)
+- [Go 1.16.5](https://github.com/golang/go/releases/tag/go1.16.5)
+
+### Bug fixes and minor changes
+
+- Prevent `docker run` from hanging if inotify event injection fails. Fixes [docker/for-mac#5590](https://github.com/docker/for-mac/issues/5590).
+- Fixed error showing stderr log in the UI. Fixes [docker/for-mac#5688](https://github.com/docker/for-mac/issues/5688).
+- Fixed an issue which caused `riscv64` emulation to fail on Docker Desktop. Fixes [docker/for-mac#5699](https://github.com/docker/for-mac/issues/5699).
+- Automatically reclaim space after deleting containers by deleting volumes and removing build cache.
+- Docker Desktop now respects the bypass proxy setting through the **Manual proxy configuration** option when pulling and pushing images. See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
```
Comment (djs55) [inline docker-for-mac/release-notes.md:51]: It's a more minor change:
```suggestion
- Docker Desktop now allows a blank HTTP proxy to be configured for the Docker engine, which will completely disable the internal HTTP proxy.  See [docker/for-mac#2467](https://github.com/docker/for-mac/issues/2467).
```

File: docker-for-mac/release-notes.md [hunk 14]
```diff
@@ -13,35 +13,79 @@ This page contains information about the new features, improvements, known issue
...
+- Docker Compose applications with file names other than `docker-compose.yml` can now be removed from Docker Desktop. Fixes [docker/for-win#11046](https://github.com/docker/for-win/issues/11046)
+- Docker Desktop now exposes the host CPU on Apple silicon. Fixes [docker/for-mac#5681](https://github.com/docker/for-mac/issues/5681).
+- Avoid leaking open ports bound to privileged ports and specific IPs over engine Restart. Fixes [docker/for-mac#5649](https://github.com/docker/for-mac/issues/5649).
+- Use `vpnkit` with `virtualization.framework` to fix connectivity issues with VPN clients such as Cisco AnyConnect.
+- Fixed version number missing in update dialog window.
+- Fixed an issue where the diagnostics were sometimes not uploaded correctly from the **Support** dialog.
+- Fixed DNS entries for `*.docker.internal` and Kubernetes cluster reset after the VM IP changes.
+
```
Comment (djs55) [inline docker-for-mac/release-notes.md:59]:
```suggestion

### Known issues
- On Apple Silicon in native `arm64` containers, older versions of `libssl` in `debian:buster`, `ubuntu:20.04` and `centos:8` will segfault when connected to some TLS servers, for example `curl https://dl.yarnpkg.com`. The bug is fixed in newer versions of `libssl` in `debian:bullseye`, `ubuntu:21.04` and `fedora:35`.
```

File: docker-for-mac/release-notes.md [hunk 15]
```diff
@@ -13,35 +13,79 @@ This page contains information about the new features, improvements, known issue
 
 {% include eula.md %}
 
-## Docker Desktop 3.3.3
-2021-05-06
+## Docker Desktop 3.4.0
+2021-06-07
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:17]:
```suggestion
2021-06-08
```

File: docker-for-windows/release-notes.md [hunk 16]
```diff
@@ -13,17 +13,55 @@ This page contains information about the new features, improvements, known issue
 
 {% include eula.md %}
 
+## Docker Desktop 3.4.0
+2021-06-07
```
Comment (usha-mandya) [inline docker-for-windows/release-notes.md:17]:
```suggestion
2021-06-08
```

File: docker-for-windows/release-notes.md [hunk 17]
```diff
@@ -13,17 +13,55 @@ This page contains information about the new features, improvements, known issue
...
+- Fixed version number missing in update dialog window.
+- Fixed an issue where the diagnostics were sometimes not uploaded correctly from the **Support** dialog.
+- Fixed DNS entries for `*.docker.internal` and Kubernetes cluster reset after the VM IP changes.
+- Fixed a corrupted internal cache which was preventing Docker Desktop from starting. Fixes [docker/for-win#8748](https://github.com/docker/for-win/issues/8748).
+
 ## Docker Desktop 3.3.3
 2021-05-06
 
-> [Download](https://desktop.docker.com/win/stable/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64)
+> [Download](https://desktop.docker.com/win/stable/amd64/64133/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64)
```
Comment (usha-mandya) [inline docker-for-windows/release-notes.md:57]:
```suggestion
[Download](https://desktop.docker.com/win/stable/amd64/64133/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64){: .button .primary-btn}
```

File: docker-for-windows/release-notes.md [hunk 18]
```diff
@@ -13,17 +13,55 @@ This page contains information about the new features, improvements, known issue
 
 {% include eula.md %}
 
+## Docker Desktop 3.4.0
+2021-06-08
+
+> [Download](https://desktop.docker.com/win/stable/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64)
```
Comment (usha-mandya) [inline docker-for-windows/release-notes.md:19]:
```suggestion
[Download](https://desktop.docker.com/win/stable/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64){: .button .primary-btn}
```

**Review by stephen-turner — APPROVED** [review]
LGTM

**Review by StefanScherer — APPROVED** [review]
LGTM

**Review by usha-mandya — APPROVED** [review]
LGTM

**Pull request merged**